api_draft.txt

```diff
@@ -0,0 +1,128 @@
+
+################################################################################
+Arimaa Game Server JSON API
+################################################################################
+
+#TODO - see todos below
+
+All requests and responses are in JSON.
+
+Shared types:
+<Message> = {"message":<String>}
+<Error> = {"error":<String>}
+<GameID> = <Int>   #Identifier for a game
+<Auth> = <String>  #Authorization token returned from logging in
+<GameAuth> = <String>  #Authorization token returned from joining a game
+<Username> = <String>
+<Timestamp> = <Float> #Seconds since the epoch (1970-01-01 00:00 UTC) (up to however much precision the server has)
+
+===ACCOUNTS========================================================
+To register an account:
+POST /accounts/register {"username":<Username>,"email":<String>,"password":<String>}
+
+#TODO add other account facilities
+#TODO add ways of looking up users
+
+===LOGIN===========================================================
+To log in:
+POST /accounts/login {"username":<Username>,"password":<String>}
+Response if successful = {"username":<Username>,"auth":<Auth>}
+Response if not successful = <Error>
+
+To log out:
+POST /accounts/logout {"auth":<Auth>}
+Response if successful = <Message>
+Response if not successful = <Error>
+
+===GAMEROOM========================================================
+<Board> = {
+  "history":[<String>,<String>,<String>,...], #Move history in the order [1g,1s,2g,2s...] ex: ["Ra1","Ra7","Ra1n Ra2n Ra3n Ra4n"]
+  "toMove":<String>,  #Next player to move, either "g" or "s"
+  "position":<String> #64-char string specifying the board in the order a8-h8,a7-h7,...,a1-h1
+                      #ex:"rrrrrrrrchdemdhc                                CHDMEDHCRRRRRRRR"
+}
+
+<UserInfo> = {
+  "username":<Username>,
+  "rating":<Float>
```

Comment:
The Free Internet Chess Server maintains a half-dozen ratings for each player:
lightning/blitz/standard: standard chess with three different time controls
Also every supported variant has its own rating.

I agree with Aaron that it wouldn't make sense to have EEE games share ratings with standard games.  (I wonder how ratings would work for handicapped games?)